### ISSUE TITLE
ADDRESS column is empty in "kubectl get ingress"

### ISSUE THREAD

**Issue by infiz** [nginxinc/kubernetes-ingress#204]
I created a local k8s cluster with kubeadm and using weave as CNI.  I would consider it as a bare-metal environment that I am adding more nodes (VM) by "kubeadm join" as well.
I install nginx-ingress using daemonset with "hostNetwork: True" so I can expose port 80 and 443 directly from the selected nodes.
Most things work as expected.  However, when I run  "kubectl get ingress", there is no IP shows up in ADDRESS column.

**Comment by pleshakov** [comment]
@infiz Unfortunately, there is no support for this yet.

**Comment by infiz** [comment]
Thanks.  Is it in the road map to support it?

**Comment by simon-k8s** [comment]
That would be great if the IP would show up. The external IP is a requirement for external dns:

https://github.com/kubernetes-incubator/external-dns/

**Comment by pleshakov** [comment]
It is on the roadmap, but I cannot provide any ETA.

**Comment by teotia39** [comment]
The same is mine requirement too . If we cannot get an external address , than why we have given ingress resource in openshift origin , Because its primary function of ingress to give external IP's.

So do we have any way in openshift origin multimaster environment where I can give external accessible ip for my services ?????

The external IP's which we get in case of serviceType loadbalancer are VIP's and a route has to be created via nodes , so in case of high availability I cannot create route via single master or a node , becasue what if that node goes off , Than route has to be created via multiple nodes for HA ..  

Comments ??

**Comment by aledbf** [comment]
> Because its primary function of ingress to give external IP's.

Ingress is not the component that assigns the IP address. If you use a service type=LoadBalancer kubernetes creates the required resourced in the cloud provider and that assigns the IP or FQDN.
If you run k8s in baremetal there's no such feature.

**Comment by nrobert13** [comment]
@infiz , if you run the controller with the option --report-node-internal-ip-address=true, you will get the your node IP's in the addresses column, so running the controler in hostNetwork mode, is exactly what you need, I guess. See my issue: https://github.com/kubernetes/ingress-nginx/issues/1750

**Comment by simon-k8s** [comment]
@nrobert13 are your sure this is possible? i just gave it a try with an nginxplus ingress. The option report-node-internal-ip-address is not defined:

``
kubectl logs nginx-plus-ingress-controller-mzdax
flag provided but not defined: -report-node-internal-ip-address
Usage of /nginx-ingress:
``

The plus controller will not start with this option.

**Comment by nrobert13** [comment]
yes, I use this image quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.9.0-beta.17 and it works like I pasted in the linked issue in my previous post.

**Comment by simon-k8s** [comment]
@nrobert13 ok that seems to be the problem. I built my own image base on these instructions:

https://github.com/nginxinc/kubernetes-ingress/tree/master/nginx-controller

The option to show the external ip doesn't seem to exist here.

**Comment by pleshakov** [comment]
@nrobert13 @simon-k8s the feature -- reporting the IP address(es) of the Ingress controller in the ADRESS column is not supported with this Ingress controller. `quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.9.0-beta.17 ` refers to [this](https://github.com/kubernetes/ingress-nginx/) Ingress controller, which supports that feature.

**Comment by nrobert13** [comment]
@pleshakov you are right. I didn't realize that are 2 different implementations of the nginx controllers....
